File: task-tracker-backend/src/main/java/ru/powergleb/task/tracker/backend/security/repositories/UserRepository.java
package ru.powergleb.task.tracker.backend.security.repositories;

import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.powergleb.task.tracker.backend.security.model.User;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Integer> {

    @EntityGraph(type = EntityGraph.EntityGraphType.FETCH, attributePaths = "tasks")
    Optional<User> findByEmail(String email);

    @EntityGraph(type = EntityGraph.EntityGraphType.FETCH, attributePaths = "tasks")
    Optional<User> findById(int userId);
}

File: task-tracker-scheduler/src/main/kotlin/ru/powergleb/task/tracker/scheduler/repositories/UserRepository.kt
package ru.powergleb.task.tracker.scheduler.repositories

import ru.powergleb.task.tracker.scheduler.model.User
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Int> {
    @EntityGraph(type = EntityGraph.EntityGraphType.FETCH, attributePaths = ["tasks"])
    override fun findAll(): MutableList<User>
}